fontforge/python.c

```diff
@@ -7242,6 +7242,62 @@ static int PyFF_Glyph_set_vhints(PyFF_Glyph *self,PyObject *value, void *UNUSED(
 return( PyFF_Glyph_set_hints(self,true,value));
 }
 
+static PyObject *PyFF_Glyph_get_user_decomp(PyFF_Glyph *self, void *UNUSED(closure)) {
+    PyObject *ret;
+
+    if ( self->sc->user_decomp==NULL ) {
+        return( Py_BuildValue("s", "" ));
+    }
+    else {
+        char* out = u2utf8_copy(self->sc->user_decomp);
+        ret = PyUnicode_DecodeUTF8(out, strlen(out), NULL);
+        free(out);
+        return ret;
+    }
+}
+
+static int PyFF_Glyph_set_user_decomp(PyFF_Glyph *self,PyObject *value, void *UNUSED(closure)) {
+    char *newv;
+    PyObject *temp;
+    unichar_t *udbuf;
+
+    if (value == Py_None) {
+        if (self->sc->user_decomp != NULL) { free(self->sc->user_decomp); }
+        self->sc->user_decomp = NULL;
+        return 0;
+    }
+
+    if ( PyUnicode_Check(value)) {
+        /* Need to force utf8 encoding rather than accepting the "default" */
+        /*  which would happen if we treated unicode as a string */
+        temp = PyUnicode_AsUTF8String(value);
+        newv = copy( PyBytes_AsString(temp));
```

Comment:
Get rid of `newv`, just assign to `udbuf` directly here (by calling `utf82u_copy`), otherwise you're leaking `newv` down the line as you never free it